### ISSUE TITLE
Enhancement request - `value in (..., )` type narrowing

### ISSUE THREAD

**Issue by Andrej730** [microsoft/pyright#10802]
[pyright-play](https://pyright-play.net/?code=CYUwZgBALiDOUApgGMoC4LAJaoNrwCcAaCQiAHwgDkB7AOxAF0BKCAWgD5r6Q0AoCIIgA3AIYAbAK4gIAXkyoAdAHMQiAEQx46kor3MBQrJDFSZWOhAS0Gu-fyGOIAYhETpGAKLjxWAA6wWLAAKgCefjKUNiAkyDQEBCCohk4QicIgEgD6UOEgCKbSBo4g4rC8KU6uhbykUAQUEN6%2BAUFhEZWOriAAHhGoIMAYhJ1C6ZniOXkF7iDMQA)

```python
def test(dct: dict[str, str | None]) -> None:
    value = dct.get("test", ...)
    if value in (None, ...):
        # value: EllipsisType | None, correct
        reveal_type(value)
    else:
        # value: str | EllipsisType
        # expected: str
        reveal_type(value)
```

**Comment by erictraut** [comment]
We do occasionally add support for new patterns, but each one requires special-case logic and has the potential for slowing down overall type analysis, so we do so only in cases where we have strong signal from pyright users that it's a common pattern. 

I've never seen this pattern before, and none of the other type checkers support it either — which supports the idea that it's not a common pattern. I'm therefore going to reject the enhancement request. 

I might reconsider in the future if there's sufficient signal from pyright users (in the form of upvotes on this issue), but for now, I'm going to close.
